### PR TITLE
Load symbols off the main thread

### PR DESCRIPTION
**Pull request by dimitry-**
With line number symbol loading is getting slower and for
large binaries when the main thread is in suspension for long
enough the tunnel generates socket error on the next request
(it could have something to do with tcp timeout).

With recent code refactoring it now possible to do symbol loading off
the main thread.

Test: Start Orbit, load symbols see that loading is done off
      the main thread.
Bug: http://b/156460883

### CODE REVIEW

File: OrbitGl/App.cpp [hunk 1]
```diff
@@ -843,38 +843,41 @@ void OrbitApp::LoadSymbols(const std::filesystem::path& symbols_path,
...
+      process->AddFunctions(module->m_Pdb->GetFunctions());
+      LOG("Loaded %lu function symbols for module \"%s\"", symbols.symbol_infos().size(),
+          module->m_FullName);
+
+      // Applying preset
+      if (preset != nullptr) {
+        auto it = preset->preset_info().path_to_module().find(module->m_FullName);
+        if (it != preset->preset_info().path_to_module().end()) {
+          for (const FunctionInfo* func : module->m_Pdb->GetSelectedFunctionsFromPreset(*preset)) {
+            SelectFunction(*func);
```
Comment (florian-kuebler) [inline OrbitGl/App.cpp:866]: I believe, selecting functions require to be done on the main thread.

**Review by florian-kuebler — COMMENTED** [review]
Thanks for the change. As you said, with the recent changes, it should be fine to load the symbols off the main thread. However, selecting functions currently assumes to be done on the main thread. Easy solution would be to schedule the apply preset part again on main thread.

File: OrbitGl/App.cpp (hunk 1 above)
Comment (dimitry-) [reply OrbitGl/App.cpp:866]: it is done on the main thread - the only part which is done of the main thread is line 850

Comment (dimitry-) [reply OrbitGl/App.cpp:866]: all legacy structures - like Module and Process are also modified on the main thread. Only loading from the elf-file has moved to thread-pool.

Comment (florian-kuebler) [reply OrbitGl/App.cpp:866]: Ah thanks. I missed line 852. Perfect. Then it seems to be all fine for me.

**Review by florian-kuebler — APPROVED** [review]

**Comment by florian-kuebler** [comment]
LGTM, thanks for the explanation and for that change.

**Pull request merged**